pkg/proc/interface.go

```diff
@@ -114,6 +114,7 @@ type BreakpointManipulation interface {
 // CommonProcess contains fields used by this package, common to all
 // implementations of the Process interface.
 type CommonProcess struct {
+	goIDToIdx     map[int]uint64
```

Comment:
Maybe we have a new type here.

Something like (and this is just a sketch):

```
type GoroutineCache struct {
  allgAddr uint64
  allg []*G
  idx  map[int]uint64
}

func (gc *GoroutineCache) AllGAddr() uint64 {
  return gc.allgAddr
}

func (gc *GoroutineCache) AddGoroutine(...) {
  ...
}

func (gc *GoroutineCache) FindGoroutine(...) *G {
 ...
}
```

That way we can abstract it a bit more and callers don't need to know the internals of where the idx map is allocated.